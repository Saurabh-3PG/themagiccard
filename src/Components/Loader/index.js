import React from "react";
import { Grid } from "@material-ui/core";
import PropTypes from "prop-types";

export default function Loader(props) {
  const { componentID } = props;
  const uniqueId = componentID + "_loader";
  return (
    <>
      <Grid
        container
        direction="row"
        alignItems="center"
        justify="center"
        className={`${uniqueId} loader`}
        id={uniqueId}
      >
        <div className="loader_icon_container">
          <div className="loader_icon" />
        </div>
      </Grid>
    </>
  );
}
Loader.propTypes = {
  componentID: PropTypes.string.isRequired,
};
