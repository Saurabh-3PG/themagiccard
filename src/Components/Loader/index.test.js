import React from 'react';
import renderer from 'react-test-renderer';
import Loader from './index';

test('Loader app component', () => {
  const component = renderer.create(
    <Loader
      componentID={'Loader'}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});