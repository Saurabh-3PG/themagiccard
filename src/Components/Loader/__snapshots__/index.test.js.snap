// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader app component 1`] = `
<div
  className="MuiGrid-root Loader_loader loader MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  id="Loader_loader"
>
  <div
    className="loader_icon_container"
  >
    <div
      className="loader_icon"
    />
  </div>
</div>
`;
